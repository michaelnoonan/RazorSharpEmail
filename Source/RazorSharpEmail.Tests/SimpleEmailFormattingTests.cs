using ApprovalTests.Reporters;
using Email.Models;
using NUnit.Framework;

namespace RazorSharpEmail.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class SimpleEmailFormattingTests
    {
        private IEmailFormatter _emailFormatter;
        private LanguageScope _languageScope;

        [TestFixtureSetUp]
        public void TestFixtureSetUp()
        {
            _emailFormatter = Given.AnInitializedEmailFormatter();
            _languageScope = new LanguageScope("en-AU");
        }

        [TestFixtureTearDown]
        public void TestFixtureTearDown()
        {
            _languageScope.Dispose();
        }

        [Test]
        public void Given_a_model_the_email_should_format_the_email_correctly()
        {
            var email = _emailFormatter.BuildTemplatedEmailFrom(
                new SimpleEmailModel
                {
                    RecipientFirstName = "Michael",
                    ReferenceNumber = "REF123456",
                    Message = "Hello World!",
                    Url = "http://google.com"
                });

            ApprovalTests.Approvals.Verify(email.Everything());
        }
    }
}