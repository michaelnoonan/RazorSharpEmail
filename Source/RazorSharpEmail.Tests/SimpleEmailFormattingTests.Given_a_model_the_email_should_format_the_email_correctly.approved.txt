Test Email: REF123456
---------------------------------
<h1 style="color: red">Alert!!!</h1>
<h3>Your reference: REF123456</h3>
<p>Dear Michael,</p>
<p>Hello World!</p>
<p>Thanks!</p>
---------------------------------
Alert!!!
Your reference: REF123456
Dear Michael,
Hello World!
Thanks!